tsfresh/feature_extraction/feature_calculators.py

```diff
@@ -2212,3 +2214,69 @@ def benford_correlation(x):
     # np.corrcoef outputs the normalized covariance (correlation) between benford_distribution and data_distribution.
     # In this case returns a 2x2 matrix, the  [0, 1] and [1, 1] are the values between the two arrays
     return np.corrcoef(benford_distribution, data_distribution)[0, 1]
+
+
+@set_property("fctype", "combiner")
+def matrix_profile(x, param):
+    """
+    TODO: Documentation
+    :param x: the time series to calculate the feature of
+    :type x: numpy.ndarray
+    :param param: contains dictionaries {"sample_pct": x, "threshold": y, "feature": z}
+    with sample_pct and threshold being parameters of the matrixprofile
+    package https://matrixprofile.docs.matrixprofile.org/api.html#matrixprofile-compute
+    and feature being one of "min", "max", "mean", "median", "25", "75"
+    and decides which feature of the matrix profile to extract
+    :type param: list
+    :return: the different feature values
+    :return type: pandas.Series
+    """
+    x = np.asarray(x)
+
+    def _calculate_mp(**kwargs):
+                """Calculate the matrix profile using the specified window, or the maximum subsequence if no window is specified"""
+        try:
+            if 'windows' in kwargs:
+                m_p = mp.compute(x,**kwargs)
+
+            else:
+                m_p = mp.algorithms.maximum_subsequence(x, include_pmp=True)['pmp'][-1]
```

Comment:
In this case the `kwargs` are not used in the function call. Is this on purpose?

Reply:
He should be using the threshold parameter here.

Reply:
@tylerwmarrs I don't think threshold needs to be set here, as we planned to go with the default value of 0.95, and that's already set in the maximum_sequence function. That said, I'll re-insert kwargs for maximum flexibility.